rust/src/schema.rs

```diff
@@ -97,54 +229,51 @@ impl Field {
             "double" => DataType::Float64,
             "binary" => DataType::Binary,
             "string" => DataType::Utf8,
-            _ => panic!(),
+            x => DataType::Extension(
+                "not_supported_yet".to_string(),
+                Box::new(DataType::Binary),
+                Some(x.to_string()),
+            ),
         }
     }
 
     /// Return Arrow Data Type name.
     pub fn type_str(t: &DataType) -> String {
```

Comment:
So this method is all about `DataType`, but nothing about `Field`.  It should not belong to `Field`?